### DISCUSSION TITLE
Some YouTube videos are detected while others aren't on 9.0.2.12 stable

### DISCUSSION THREAD

**Discussion by xecus922** [aclap-dev/video-downloadhelper#942] [Help and Q&A]
### Your question

Windows 10 Pro 64-Bit (10.0, Build 19045)
VDH version 9.0.2.12 Stable
CoApp version v2.0.19
Firefox version 128.08b (64-Bit) Beta 

Looks like some YouTube videos aren't detected while others are, I just noticed this today. 

**Undetected** 
https://www.youtube.com/live/PDkK_rrlAM8
https://www.youtube.com/live/Hz72l8mxzbs
https://www.youtube.com/live/DU8bpHSq1tE

**Detected**
https://www.youtube.com/watch?v=juQL-0db_wc 
https://www.youtube.com/shorts/SUgK4k7A9ic
https://www.youtube.com/watch?v=ARYM9ebZ6r8

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "D:\\Download",
    "concurrent_downloads_max": 10,
    "never_show_no_incognito_msg_again": true,
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "low_quality": true,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "yt_warning": false,
    "use_sidebar": false,
    "open_count_store": 11,
    "successfull_dl": 59
}

**Comment by umbuzada** [comment]
I'm having this exact same issue.

**Comment by Abelius1** [comment]
I'm also experiencing this issue.
FYI, it can detect MKV videos, but I've stopped seeing the WebM videos I was seeing for 4K videos on YT.

**Reply by GrampaWildWilly** [reply]
@xecus922's second URL in his undetected group has a 2160p MKV variant . . . that is, after you apply Matt's slick trick.  Some content has a 4K variant, some doesn't.  That's entirely at the discretion of whoever uploads the content.

WEBM is a subset of MKV.  MKV is a broader type of container.  @paulrouget Paul has discarded offering WEBM variants in favor of offering MKV instead.  There is no loss of functionality.  YouTube has a rather bizarre way of offering their content.  VDH now detects everything that is there, which it did not necessarily in the past.  And by past, I mean versions before VDH 9.  So WEBM is now shown as MKV, which is a broader specification, & all the variants are now offered.  If a 4K variant is available, VDH will show it.  If you don't see it in the VDH menu, that's because the author of the content didn't provide a 4K variant of the content.  A 4K variant is not on YouTube so VDH isn't going to manufacture a 4K variant out of nothing.

**Reply by Abelius1** [reply]
> @xecus922's second URL in his undetected group has a 2160p MKV variant . . . that is, after you apply Matt's slick trick. Some content has a 4K variant, some doesn't. That's entirely at the discretion of whoever uploads the content.

This would be fine, if not for the small detail that EVERY single content creator that was uploading videos a couple of days ago, has stopped being detected at the same time, overnight.
I'm pretty sure they didn't have an international meeting on Zoom and said "Hey, let's start uploading in another format tomorrow'.

**Reply by GrampaWildWilly** [reply]
Well, I don't pretend to have looked at the output of "every single content creator" or even a lot of them.  I basically go to YouTube a couple of times a week for content that I am interested in.  Otherwise, the only reason I visit the site is to investigate issues with videos my fellow users are reporting.  I'm sure you have a list of your favorite content creators who suddenly stopped being detected properly.  Let's settle on 3 of them, to keep things manageable.  Let's have 6 URLs, 2 from each creator.  Show one URL that's old enough it has 4K content & one that's newer that you say you are not seeing 4K content on.  We need to apply Matt's trick to the URL, if necessary, before we end investigation of any individual URL.  We need to compare the list of offered resolutions you can see from the gear icon inside the YouTube player on the web page against the list of variants being offered on the VDH menu.  I think VDH will have more entries because the typical situation on YouTube is that they offer both MP4 & MKV variants at most resolutions.  But when there is content at resolutions higher than 1080p, those are usually offered only as MKV.  The gear icon will show only the resolution but VDH will show both an MP4 & an MKV variant at that resolution.  Another rather strange thing is that VDH usually recognizes 2 or 3 extra MKV variants at the very lowest resolution, like 240p or 144p.  This is due to that bizarre method I mentioned earlier of how YouTube presents its content.

But before you do that, update your VDH to the beta.  I'm using the beta & I saw some 4K content on one of your URLs that you provided in your undetected list, but not on the other 2.  So let's at least make sure you are not encountering some bug that Paul has fixed.

**Comment by Mattmjs** [comment]
I see, are they testing a new link format ? What's the deal with having **live** in the link if it's not a live stream.
You can resolve it by replacing **live/** with **watch?v=** in the YouTube link.

**Reply by xecus922** [reply]
> I see, are they testing a new link format ? What's the deal with having **live** in the link if it's not a live stream. You can resolve it by replacing **live/** with **watch?v=** in the YouTube link.

Seems so. 

I actually watched a link change back from the 'live' to 'watch?v=' within the span of 6 to 8 hours or so, downloaded it before I knew about manually changing it and now it's back to 'live' again. But your solution definitely works thank you. 

Not sure if I should mark as answer because while this workaround definitely works, we don't know for how long and many others who don't even come on here will continue to face the issue.

**Reply by Abelius1** [reply]
> I actually watched a link change back from the 'live' to 'watch?v=' within the span of 6 to 8 hours or so, downloaded it before I knew about manually changing it and now it's back to 'live' again. But your solution definitely works thank you.

https://www.youtube.com/watch?v=<video_code> links don't work for me, so no, it's not a solution.

**Reply by Mattmjs** [reply]
@Abelius1 , what is your example that didn't work.

**Reply by Abelius1** [reply]
> @Abelius1 , what is your example that didn't work.

_Update while writing this comment: I think every 'Age Restricted' video isn't getting its links detected. Which sucks because I bought the plugin to download those._

I'd say (speculation) that every video with a WebM format in one of their available resolutions isn't being detected, regardless of whether the lower resolutions aren't WebM. The only one I was able to download at 4K had an MKV extension instead of WebM.

I'm using the latest Edge version (126.0.2592.81) and licensed VDH 9.0.2.12.

I've disabled both AdBlock Plus and uBlock Origin, although it was working with them a couple of days ago. I've also deleted all YouTube cookies and restarted.

Nothing changed version-wise. It just stopped detecting most videos overnight. This looks like YT is changing something, and VDH hasn't been adapted yet.

Examples (be advised, most undetected videos are NSFW):

NOT detected:
https://www.youtube.com/watch?v=wF1YEwFY3ZQ
https://www.youtube.com/watch?v=ogH3fQsdsSo
https://www.youtube.com/watch?v=nT5BMql2cFk
https://www.youtube.com/watch?v=LpZiXTKfIQc
https://www.youtube.com/watch?v=L2XcQAUhwa8

Detected:
https://www.youtube.com/watch?v=slKGOije-Nc
https://www.youtube.com/watch?v=bXVOATfgLW8
https://www.youtube.com/watch?v=DAZL68Ekqy8
https://www.youtube.com/watch?v=oQ7qSDW3zxw
https://www.youtube.com/watch?v=C-6FI9mLkTM&t=456s
https://www.youtube.com/watch?v=kC6CrUZcW-A
https://www.youtube.com/watch?v=f6rerulM6bc
https://www.youtube.com/watch?v=wtg7AetxuWo
https://www.youtube.com/watch?v=whJ2o6I66ds

Detected after a Ctrl+F5:
https://www.youtube.com/watch?v=jInlO6-JTww
https://www.youtube.com/watch?v=0l5ZmK08FY0
https://www.youtube.com/watch?v=S4W-aKEsHBw
https://www.youtube.com/watch?v=AtbT9-xd4p4

**Reply by Abelius1** [reply]
> _Update while writing this comment: I think every 'Age Restricted' video isn't getting its links detected. Which sucks because I bought the plugin to download those._

Yup, confirmed. EVERY video categorized as 'Age Restricted' has become non-detectable by VDH.

**Reply by GrampaWildWilly** [reply]
The Age Restricted issue is very old.  It is not something that just happened.  You can find reports of this on our old Google Groups forum going back . . . I don't know, I'm not going back to verify.  You can go back to verify this if you want because after a short spell off the air, the old Google Group is back on the air.  In case you're interested:

https://groups.google.com/g/video-downloadhelper-q-and-a

But it's been, as I recall it, at least a couple of years since people started reporting this issue.  Are you saying that until recently, you WERE able to get VDH to download Age Restricted content from YouTube?

As for MKV vs WEBM, I already explained that above.  Paul has discarded the less flexible WEBM format for MKV.  There is no difference to what you would download as WEBM vs MKV.  The WEBM format is a type of MKV.  WEBM is a subset of MKV.  Just accept that you will consistently see MKV in VDH from now on & you will not see WEBM.  But what you see that is labelled MKV is the same as what you used to see that was labelled WEBM.  You can either take my word for it or insist that Paul explain it for you in greater technical detail, something I can't do.

I visited each of your links @Abelius1 & looked at the VDH menu being offered.  But my environment is Windows 11 Pro 64-bit version 23H2 build 22631.3810, Firefox 127.0.2 64-bit, licensed VDH 9.0.2.12a1 beta, CoApp 2.0.19.  Firefox, not Edge.  And the beta version of VDH, not the general availability version.

For your "NOT detected" category, I logged in with my free Google account.  YouTube showed me the video.  None of these videos showed the DRM indicator.  VDH detected nothing, even after a Ctrl+F5 page reload.  This confirms not only your observation but the very old issue I already mentioned.  After the last of those URLs, I logged out of Google.

For your "Detected" category, I confirm that every single one of them was correctly detected by my VDH beta for Firefox.  In addition, I dropped down the list of variants on each VDH menu.  Some of the videos did not show resolutions higher than 1080p, so the list of MP4 variants & the list of MKV variants was more or less the same.  In some cases, the MKV variants included multiple 1080p variants with different bit rates, indicating the availability of different quality levels within the same resolution.  Those multiple quality variants were available only as MKV, not MP4.  Some of the videos showed MKV variants at 2160p & 1440p that were not shown as MP4 variants.  This is to emphasize something I said upthread here.  Some videos offer 4K variants & some don't.  That's on the author of the content.  VDH detects what's there.

For your "Detected after a Ctrl+F5" category, I found that I didn't need to reload the page.  VDH detected the content on first page load.  But like I've said, I'm using Firefox & the VDH beta.  I believe this is an issue with the browser, not VDH.  But I could be educated to learn that it's an issue between the general availability version vs the beta.  I think that if you do ever have to reload a page to make VDH recognize content, that's fine.  If it recognizes content, you should be happy, no matter whether you have to reload the page.  That has always been the way VDH works.  The warning about clicking play is still on the VDH menu.  Maybe the warning should add "or reload the page" to its wording.  But I say the necessity of reloading some given page is not a bug in VDH.  It's a bug when reloading doesn't make VDH recognize content, as is the case for Age Restricted material.  I have always said that is a bug.  I believe Paul said at some point he was looking into this Age Restricted issue, but I could be wrong.

**Reply by Abelius1** [reply]
> But it's been, as I recall it, at least a couple of years since people started reporting this issue. Are you saying that until recently, you WERE able to get VDH to download age-restricted content from YouTube?

That's correct. I've been downloading Age Restricted videos from YT for about a week, without a full VDH license, then I decided to buy. Then I downloaded more without restrictions, for 2-3 days, and this issue started to happen.

In other words, I wasn't aware that age-restricted videos were considered a 'known issue' beforehand. For me, the plugin simply has stopped working as expected from my previous experience.

If you, or the developer, confirm this is a known issue that won't ever be addressed, I guess I'll need to ask for a refund.

**Reply by GrampaWildWilly** [reply]
I can't confirm what will or won't happen in the product.  I don't work for VDH.  I don't have any inside information.  Paul will or won't do whatever he decides.  My input carries no more weight than any other user's input.  All I know is a lot of people have reported problems downloading from YouTube when they are logged in on YouTube.  That would be a login with an ID for which they paid.  I have been able to download things from YouTube on the rare occasions when I visit YouTube & log in with my free Google ID, which is the only one I have.  However, I have not been able to download Age Restricted content, which necessarily means being logged in, & for me, VDH didn't recognize any of your Age Restricted videos.  I even went to the extent of unregistering my VDH license, closing all browser windows, & trying again.  VDH still didn't recognize the Age Restricted videos.  There have been reports, quite infrequent, of YouTube videos that VDH does not recognize & those are not Age Restricted.  So maybe there's just one problem with videos that are presented a certain way.  Or maybe there's 2 problems, one being the certain presentation & the other being whether you're logged in.  As just another user, I can't tell.

I will say that one of your Age Restricted videos really shouldn't be.  It's the interview highlight with Violet Myers.  I know about that one because Holly Randall Unfiltered is one of those things that gets me to visit YouTube not in connection with other users' problems.  Those interviews have always been available, not Age Restricted, not requiring a login of any kind, paid or otherwise.  I downloaded & watched Violet's interview when it was first posted 2 years ago (https://www.youtube.com/watch?v=4OtsusqbzZQ&list=PLaMF_frL0_EzL3NmqZF65G60wzF4YCNJk).  So the full item is available to anybody without the need to log in.  It's just dumb that an extract from that full video would get the Age Restriction.  That is only tangentially related to this discussion.  The Holly Randall Unfiltered interviews do all download just fine with VDH & have done so for years, with the occasional interruption while VDH needed repair due to one of YouTube's frivolous changes to their operation.  When I say years, I was downloading those interviews with VDH 7 & VDH 8, many releases & beta releases of each.  VDH 9 continues to download those, the most recent time being this past Sunday.  I anticipate no problems with the one on the schedule for this coming Sunday.

However, I must point out again that I am using Firefox & the VDH beta.  Maybe using the general availability release would make a difference.  Maybe not having a VDH license makes a difference.  Maybe things work differently on Edge.  I am not about to test any of those conjectures, although I did try removing my license & VDH still didn't recognize any of the Age Restricted content.  I find it extremely weird that, for you, it would work for a time while you did not have a VDH license,  & then it still worked for a time after you got a VDH license, but then it stopped working.  Did the Age Restricted content you successfully downloaded have the watermark before you got your license?  Did it take an inordinate amount of time to write the watermarks into those items?  I assume that for the couple of days it worked after you got your license, you got no watermarks.

It's up to you whether you ask for a refund.  If you bought your license within the past 30 days, & it sounds like you did, @jcfrog Jérôme will gladly refund your purchase.  But you have to explicitly ask for that in the relevant part of this forum.  Your comment here does not constitute a request for a refund.

**Reply by Abelius1** [reply]
> I will say that one of your Age Restricted videos really shouldn't be.

That was just an example to confirm what I observed about age-restricted videos being a problem.

> I find it extremely weird that, for you, it would work for a time while you did not have a VDH license, & then it still worked for a time after you got a VDH license, but then it stopped working.

I've got no idea why. I'm positive I didn't change my MO from one day to another, and no updates were made to the software. My bet is on YT changing something, because when I return to old videos that were detected, they aren't anymore.

> Did the Age Restricted content you successfully downloaded have the watermark before you got your license?

I don't have any watermarks on the videos before the licensing. To the best of my knowledge, they only get added when converting from one format to another, no? I just downloaded them as they were.

> It's up to you whether you ask for a refund. If you bought your license within the past 30 days, & it sounds like you did, @jcfrog Jérôme will gladly refund your purchase. But you have to explicitly ask for that in the relevant part of this forum. Your comment here does not constitute a request for a refund.

Yes, will do. Thank you for your help. Even though I kind of expected a developer to chime by now, in a post with this traffic.  :-/  That's basically what makes me think of refunding, not thinking the software isn't useful or the dev doesn't deserve to be paid (I'm a game dev myself).

**Reply by GrampaWildWilly** [reply]
This thread contains a lot of talk about the VDH license:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/12-fRkVJ8qo

Almost all of that thread still applies today.  Within that thread in a post dated 2022/8/11 our fellow user mjs (now going by the name @Mattmjs) explained when the watermark would be added.  All YouTube content used to get the watermark.  I experienced that back in the days before I bought my license.  That's got to be at least 5 years ago, maybe longer.

But there's been a lot of change in the 2 years since Matt explained those things.  The major change is that VDH did not used to use ffmpeg for downloads & now it does.  Aggregation of video & audio streams used to be a separate step that occurred after the data transmission.  Now, VDH does use ffmpeg for downloads & ffmpeg automatically aggregates video & audio on the fly.  So aggregation is not a separate step any more.  Maybe the watermark isn't written in cases where it used to be.  I don't know since I've had a license for so long I usually forget the watermark can even be an issue.  Conversion is one instance in which you would get a watermark but it was never the only instance.

**Reply by redmondgiant** [reply]
Can someone help me understand the specific issue of  "No media to process in the current tab" empty lists? I use Premium VDH and also see this behavior. I also see no downloads and I assume other Premium VDH users see an empty list too.

1. Are no choices received by VDH from the Youtube API, or does VDH's internal behavior/decisions result in none of what is returned being offered?

2. Is VDH using the Youtube API in a extension/CoApp/process context that can't (or won't) use the page's logged in state or cookie access, such as behaving as an anon request would even if the page doesn't? In particular, would it require the VDH context to "log in again on its own" and it can't or doesn't want to do that?

3. Is FF running in a separate process and the way in which the content data is transmitted is now different and subject to limitations it didn't have when it was separately downloaded first just to get it out of Youtube storage? If FF is used to directly download then doesn't it have to somehow have creds available, and if that is too hard or a problem then can VDH just return to locally staging the content data first (even if this results in some kind of download size limits, it beats seeing nothing as an option)?

4. Bottom Line: How does Firefox play the content fine, yet no download links are offered by VDH (if not explained above)?

**Reply by Mattmjs** [reply]
I thinks the devs could answer these questions but I'm a user like you. Have you tried viewing youtube video without Premium
to get an idea if there is detection. If so that would suggest it's something specific to the premium service maybe like private
youtube video. I found that on private videos they don't offer HLS stream versions only the adaptive formats which VDH used
to offer but now discarded. It's now only HLS formats. That's my take on it.

**Comment by paulrouget** [comment]
> Bottom Line: How does Firefox play the content fine, yet no download links are offered by VDH (if not explained above)?

Ah that's because FF doesn't just tell us "hey, there's a video here". We need to use tons of (complex) heuristics to detect videos, in a generic manner. And apparently, these heuristics don't work for these specific videos.

I'll look into it.

**Comment by kn1ghtofcups** [comment]
I think I may be having a similar issue. 

Some Youtube videos are recognised by VDH just fine, others I can't get it to recognise at all. For the one I'm currently struggling with, VeeVee on Opera detected it (but wouldn't download, presumably because it's YouTube), so I think it's something to do with the Firefox version of VDH (I have a license). I tried installing the beta version but it didn't help.

I'm not tech savvy at all (my computer might as well be a magic box) so any layman's guidance would be appreciated. Normally I'd just look elsewhere but I have been unable to locate this video anywhere else.

My info is below, and this is the video that I can't get VDH to detect - https://www.youtube.com/watch?v=cVGTRyCZmk0

version: 9.0.2.12
target: mozilla
channel: stable
lang: en-GB
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"447c2658e71b4a06"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\mxash\\Downloads",
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "low_quality": true,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 6,
    "successfull_dl": 2
}

**Reply by arnocs** [reply]
Having the same issue,

MacOS Sequoia 15.1 ARM | Firefox 132.0.2 | VDH v9.1.0.41 (beta) & v9.1.0.44 | CoApp v2.0.19

**Issue:**
The video (URL) does not get detected after starting the replay of the video, same result with opening it in a new tab.

Full URL: https://www.youtube.com/live/%youtube-video-id%
%youtube-video-id% is the youtubes video code.

**Example** URL https://www.youtube.com/live/T-xJL1T_RgQ

**Workaround:**
Copy the %youtube-video-id% in https://www.youtube.com/watch?v=%youtube-video-id%

**Note:**
It's just an URL for loading a video, probably VDH does not detect this as youtube.
Video used has in .mp4 & mkv formats

Regards, Arno

**Reply by GrampaWildWilly** [reply]
You posted

https://www.youtube.com/live/T-xJL1T_RgQ

I modified it to

https://www.youtube.com/watch?v=T-xJL1T_RgQ

Using the watch?v= trick, VDH detects content no problem on that South Park clip.  It even offers a 1440p variant in BOTH MP4 & MKV, which is unusual.  Usually, the resolutions above 1080p are offered only as MKV & not MP4.  But here I'm getting both.  There is no 2160p variant on offer, neither as MKV nor MP4.

I don't know to what extent non-detection on /live/ URLs on YouTube is a bug in VDH.  I don't know if it's being investigated.  I don't work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 24H2 build 26100.2314, Firefox 132.0.2 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20, quite a lot of changes since this thread began.
